doc/new-guidelines.md

```diff
@@ -163,6 +168,26 @@ Use the simple `defn` to declare components. Don't use `utils.views/defview` and
     (do-something window-width)))
 ```
 
+#### Use `[]` instead of `()` in Reagent components
+
+- The `()` version [does NOT work with Form-2 and
+  Form-3](https://github.com/reagent-project/reagent/blob/master/doc/UsingSquareBracketsInsteadOfParens.md#a-further-significant-why)
+  components.
+- Components defined with `[]` will be [more efficient at re-render
+  time](https://github.com/reagent-project/reagent/blob/master/doc/UsingSquareBracketsInsteadOfParens.md#which-and-why)
+  because they're interpreted by Reagent and transformed into distinct React
+  components, with their own lifecycle.
+
+```clojure
+;; bad
+[rn/view
+ (render-parsed-text message)]
+
+;; good
+[rn/view
+ [render-parsed-text message]]
```

Comment:
btw this naming `render-parsed-text` doesn't look good, i mean its good for this example but its not really good for the example because of bad naming, idk :)

Reply:
I get what you mean. I'll pick another one since I'm at it.

Reply:
btw i just realized that most of the time this error comes from function naming, `render-parsed-text` doesn't sound like a component, it sounds like a function you need to invoke once , so devs think `()` should be used here